.github/workflows/scala.yml

```diff
@@ -1,6 +1,12 @@
 name: Scala CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - dev
```

Comment:
Why only dev ?
I would say should be for all branches ?

Reply:
The `pull_request` case after matches it

Reply:
I mean, both should be for all branches i would say ?

Reply:
In this case for PRs you get the flow run twice, displaying annotations twice and consuming twice as much energy.

If we do not care about energy we can do `if: github.event_name != "pull_request" || github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name` for the lint job to avoid annotations twice?

Ref: https://wildwolf.name/github-actions-how-to-avoid-running-the-same-workflow-multiple-times/

Reply:
Ahhh damned XD
Right, so we do care about it as the regression take quite a bit of processing power.

So the case where things run twice is when sombody do a PR, each of his commit will trigger the regression on his repo and the SpinalHDL repo ?

Reply:
No, it runs on each push, only on the last commit / top of the branch (I think we can't do anything against this, just don't `git push` if you don't want to share for a peer-review).

It is more that for a given pushed commit, the workflow is run twice, once as a `push` on the author repository, and once as a `pull_request` on the target repository. If they are the same, currently there are 2 workflows started to do exactly the same thing on exactly the same commit, which make annotations displayed twice.

With the suggested modification it will always run on `dev` (if you push on `dev` so that others don't see what you are doing XD or after merging a PR) and run on branches which have an open PR to `dev`.